wandb/sdk/artifacts/artifact_saver.py

```diff
@@ -274,12 +274,16 @@ def _cleanup_staged_entries(self) -> None:
         """Remove all staging copies of local files.
 
         We made a staging copy of each local file to freeze it at "add" time.
-        We need to delete them once we've uploaded the file or confirmed we
-        already have a committed copy.
+        If we chose to skip caching, we need to delete them once we've uploaded
+        the file or confirmed we already have a committed copy.
         """
         staging_dir = get_staging_dir()
         for entry in self._manifest.entries.values():
-            if entry.local_path and entry.local_path.startswith(staging_dir):
+            if (
+                entry.local_path
+                and entry.skip_cache
```

Comment:
I think this code doesn't need to change. We've either already copied the file to the cache and can safely delete the staging copy, or we've previously determined we're skipping the cache but still can safely delete the staging copy. My main question is whether we need this step at all, given that we're deleting the staged copy in `wandb_storage_policy.py`

Reply:
Ah i see what you mean now. I was returning early if `skip_cache=True` which i now realize wasn't necessary. I agree with you now - I think we can delete the staging cleanup method since we're deleting the staging either way as a part of `_write_cache()`